src/MediaCollections/Filesystem.php

```diff
@@ -77,6 +77,23 @@ public function copyToMediaLibraryFromRemote(RemoteFile $file, Media $media, ?st
         );
     }
 
+    protected function shouldCopyFileOnDisk(RemoteFile $file, Media $media, string $diskDriverName): bool
+    {
+        if ($file->getDisk() !== $media->disk) {
+            return false;
+        }
+
+        if ($diskDriverName !== 's3') {
```

Comment:
Checking "s3" here feels a bit strange. Wouldn't it be better to check for `=== 'local'` here?

Reply:
Yes, we could use ```=== 'local'``` here, but I did not know how that would affect other filesystem drivers performance-wise.

I assumed that copy would be most efficient for most drivers and even for 's3' if no custom headers are provided.

If you want or if I'm missing something then let me know and I can change it to ```=== 'local'```.

Reply:
@freekmurze Please do let me know what you think?

Reply:
Let's go with `=== 'local'` 👍

Reply:
Done. I also squashed the commits in the process to keep it clean. Please let me know if more work is needed or can we merge it now?